src/ws/session-utils.c

```diff
@@ -574,12 +574,11 @@ scan_btmp (const char *username,
       if (errno == ENOENT)
         {
           /* no btmp → no failed attempts */
-          success = true;
-          goto out;
+          return true;
         }
 
       warn ("open(%s) failed", _PATH_BTMP);
-      goto out;
+      return false;
```

Comment:
Nitpick: May have been more robust to conditionalize the close() on `if (fd >= 0)`, but this code path is close enough to the top so that it doesn't matter.

Reply:
I also have a strong preference for the conditionalise-`close()` approach.